src/components/AutocompleteLocation/index.tsx

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable camelcase  */
+/* eslint-disable react/function-component-definition */
+import { useEffect, FC } from 'react';
+import { useDispatch } from 'react-redux';
+import { useJsApiLoader } from '@react-google-maps/api';
+import usePlacesAutocomplete, { GeocodeResult, getGeocode, getLatLng } from 'use-places-autocomplete';
+import useOnclickOutside from 'react-cool-onclickoutside';
+// function
+import { getWeathers, getWeathersInfoByCity } from 'store/weather-service/action';
+import { setCoordinates } from 'store/location-service/action';
+// types
+import { Description, Libraries } from 'core/types';
+// components
+import ButtonGps from 'components/ButtonGps';
+// style
+import s from './AutocompleteLocation.module.scss';
+
+const libraries: Libraries = ['places'];
+const GOOGLE_API: string | undefined = process.env.REACT_APP_GOOGLE_API;
+
+const AutocompleteLocation: FC = () => {
+	const dispatch = useDispatch();
+
+	const { isLoaded } = useJsApiLoader({
+		id: 'google-map-script',
+		googleMapsApiKey: String(GOOGLE_API),
+		libraries,
+	});
+
+	const {
+		ready,
+		value,
+		suggestions: { status, data },
+		init,
+		setValue,
+		clearSuggestions,
+	} = usePlacesAutocomplete({
+		initOnMount: false,
+		debounce: 300,
+	});
+
+	useEffect(() => {
+		if (isLoaded) {
+			init();
+		}
+	}, [isLoaded, init]);
+
+	const ref = useOnclickOutside(() => {
+		clearSuggestions();
+	});
+
+	const handleInput = ({ target }: any): void => {
+		setValue(target.value);
+	};
+
+	const handleSelect =
+		({ description }: Description) =>
```

Comment:
will be better using destructing inside a function

Reply:
done